### PR TITLE
feat: support to disable feature

### PR DESCRIPTION
**Pull request by maudL1n**

### CODE REVIEW

File: pkg/backup/backup.go [hunk 1]
```diff
@@ -142,6 +143,15 @@ func (bak *Server) GetEtcdBackup(name, namespace string) (*backupapiv2.EtcdBacku
 	return bak.encodeBackupObj(obj)
 }
 
+//DeleteEtcdBackup deletes etcd backup
```
Comment (tangcong) [inline pkg/backup/backup.go:146]: // DeleteEtcdBackup deletes etcd backup

File: pkg/backup/backup.go [hunk 2]
```diff
@@ -40,6 +40,7 @@ import (
 
 	kstoneapiv1 "tkestack.io/kstone/pkg/apis/kstone/v1alpha1"
 	"tkestack.io/kstone/pkg/controllers/util"
+	util2 "tkestack.io/kstone/pkg/featureprovider/util"
```
Comment (tangcong) [inline pkg/backup/backup.go:43]: featureutil

File: pkg/controllers/etcdcluster/etcdclusters_controller.go [hunk 3]
```diff
@@ -344,12 +344,25 @@ func (c *ClusterController) handleClusterFeature(cluster *kstonev1alpha1.EtcdClu
 	}
 
 	for name := range featureprovider.EtcdFeatureProviders {
+		featureName := kstonev1alpha1.KStoneFeature(name)
+		feature, err := c.GetFeatureProvider(name)
+
 		if !c.enabledFeatureGate(annotations, name) {
+			if cluster.Status.FeatureGatesStatus[featureName] != "disabled" {
```
Comment (tangcong) [inline pkg/controllers/etcdcluster/etcdclusters_controller.go:351]: define constant string

File: pkg/featureprovider/util/util.go [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package util
+
+import (
+	"strconv"
+	"strings"
+	kstoneapiv1 "tkestack.io/kstone/pkg/apis/kstone/v1alpha1"
```
Comment (tangcong) [inline pkg/featureprovider/util/util.go:24]: import package group

	"strconv"
	"strings"

	kstoneapiv1alpha1 "tkestack.io/kstone/pkg/apis/kstone/v1alpha1"

File: pkg/monitor/prometheus.go [hunk 5]
```diff
@@ -466,6 +487,31 @@ func (prom *PrometheusMonitor) initEtcdEndpoint(cluster *kstonev1alpha1.EtcdClus
 
 // SyncPrometheusMonitor sync prometheus monitor for etcdcluster
 func (prom *PrometheusMonitor) SyncPrometheusMonitor(cluster *kstonev1alpha1.EtcdCluster) error {
+	if !prom.IsMonitorEnabled(cluster) {
```
Comment (tangcong) [inline pkg/monitor/prometheus.go:490]: func sync(xxx)
  err := cleanResourceIfDisabled(..)

  err := syncResourceIfEnabled(..)

File: pkg/backup/backup.go [hunk 6]
```diff
@@ -260,6 +275,20 @@ func (bak *Server) Equal(cluster *kstoneapiv1.EtcdCluster) bool {
 // SyncEtcdBackup synchronizes the latest backup configuration.
 func (bak *Server) SyncEtcdBackup(cluster *kstoneapiv1.EtcdCluster) error {
 	namespace, name := cluster.Namespace, cluster.Name
+	if !bak.isBackupEnabled(cluster) {
+		_, err := bak.GetEtcdBackup(name, namespace)
+		if err != nil {
```
Comment (lianghao208) [inline pkg/backup/backup.go:280]:
```go
if _, err := bak.GetEtcdBackup(name, namespace); err != nil {
     if k8serors.IsNotFound(err) {
         return nil
     }
     return err
}
```

File: pkg/backup/backup.go [hunk 7]
```diff
@@ -260,6 +275,20 @@ func (bak *Server) Equal(cluster *kstoneapiv1.EtcdCluster) bool {
...
+			if k8serors.IsNotFound(err) {
+				return nil
+			}
+			return err
+		}
+		err = bak.DeleteEtcdBackup(name, namespace)
```
Comment (lianghao208) [inline pkg/backup/backup.go:286]:
```go
if err := bak.DeleteEtcdBackup(name, namespace); err != nil {
        klog.Errorf("failed to delete etcd backup, namespace is %s, name is %s, err is %v", namespace, name, err)
	return err
}
```

**Comment by tangcong** [comment]
image

**Review by tangcong — APPROVED** [review]
thanks.

**Pull request merged**